data-model/src/main/java/io/micronaut/data/annotation/QueryResult.java

```diff
@@ -45,6 +47,11 @@
      */
     String column() default DEFAULT_COLUMN;
 
+    /**
+     * @return The JSON type that resulting column will hold. It helps to pick proper column reader based on result data type
+     */
+    JsonType jsonType() default JsonType.STRING;
```

Comment:
I think the default should be always `NATIVE`

Reply:
I agree, changed.